go/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"periph.io/x/host/v3"
+)
+
+const (
+	LEDPinName    = "18"
+	ButtonPinName = "23"
+)
+
+func main() {
+	var err error
+
+	_, err = host.Init()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	printer := NewPrinter()
+
+	// we have our context so we can terminate the program
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Handle sigterm and await termChan signal
+	termChan := make(chan os.Signal)
```

Comment:
TODO: make this buffered with `1`

This piece of code handles signal interrupts so I can shut everything down.